### DISCUSSION TITLE
I displayed the dialog box as described in the document, but it did not show up

### DISCUSSION THREAD

**Discussion by longlong9568** [kikipoulet/SukiUI#295] [Q&A] [answered]
document:

<suki:SukiWindow>
	<suki:SukiWindow.Hosts>
		<suki:SukiDialogHost Manager="{Binding DialogManager}"/>
	</suki:SukiWindow.Hosts>
<suki:SukiWindow>
public class ExampleViewModel
{
	public ISukiDialogManager DialogManager { get; } = new SukiDialogManager();
}

My code is as follows

using Avalonia.Collections;
using CommunityToolkit.Mvvm.ComponentModel;
using CommunityToolkit.Mvvm.Input;
using LDCrossPlat.Model;
using LDCrossPlat.Views;
using ReactiveUI;
using Splat;
using SukiUI;
using SukiUI.Controls;
using SukiUI.Dialogs;
using SukiUI.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using ToolCrossPlat;

namespace LDCrossPlat.ViewModels
{
    public partial class TcpUseViewModel : ViewModelBase
    {
       
        

        private readonly SukiTheme _sukitheme = SukiTheme.GetInstance();
        public IAvaloniaReadOnlyList<SukiColorTheme> AvailableColors { get; }

        public ISukiDialogManager DialogManager { get; } = new SukiDialogManager();
        
        [ObservableProperty] private bool _isStartEnable;
        [ObservableProperty] private string _strServerIp;
      
        public TcpUseViewModel()
        {
           
             AvailableColors = _sukitheme.ColorThemes;
            IsStartEnable = true;  
            StrServerIp = string.Empty;  
        }

        [RelayCommand]
        private void ShowDialog()
        {
            DialogManager.CreateDialog()
        .WithTitle("Example Dialog")
        .WithContent("The content of an example dialog can be seen here.")
        .TryShow();
        
        }
    }
}

My view is as follows 
 <Design.DataContext>
	<vm:TcpUseViewModel />
</Design.DataContext>
<suki:SukiWindow.Hosts>
	<suki:SukiDialogHost Manager="{Binding DialogManager}"/>
</suki:SukiWindow.Hosts>
<ScrollViewer VerticalScrollBarVisibility="Hidden" >
	<Grid Margin="5"    HorizontalAlignment="Stretch" VerticalAlignment="Stretch">
		<Grid.ColumnDefinitions>
			<ColumnDefinition Width="*" />

		</Grid.ColumnDefinitions>

		<Grid.RowDefinitions>
			<RowDefinition Height="*" />
		</Grid.RowDefinitions>

		<suki:GlassCard Grid.Column="0" Margin="0,0,5,0" HorizontalAlignment="Stretch" VerticalAlignment="Stretch">
			<suki:GroupBox Header="Tcp Test" Margin="0" HorizontalAlignment="Stretch" VerticalAlignment="Stretch"
						   BorderBrush="White" BorderThickness="1">
				 
					<StackPanel HorizontalAlignment="Stretch" VerticalAlignment="Stretch">
						<StackPanel Orientation="Horizontal" HorizontalAlignment="Stretch" Margin="20,10,0,0" VerticalAlignment="Stretch">
							 
							<StackPanel Orientation="Horizontal" VerticalAlignment="Center">
								<Button Content="Open"
										Width="100"
										Margin="5,0,0,0"
										Command="{Binding ShowDialogCommand }"
										IsEnabled="{Binding IsStartEnable}"
										VerticalAlignment="Center" />
							</StackPanel>
						</StackPanel>

					</StackPanel>
			</suki:GroupBox>
		</suki:GlassCard>
	</Grid>
</ScrollViewer>

When I click the Open button, no action is taken  ，What is the reason for this？

**Comment by AuroraZiling** [comment]
```xml
<!-- XMLNS definitions omitted for brevity -->
<suki:SukiWindow>
	<suki:SukiWindow.Hosts>
		<suki:SukiDialogHost Manager="{Binding DialogManager}"/>
	</suki:SukiWindow.Hosts>
<suki:SukiWindow>
```

This code should be defined in the SukiWindow and not in any other Views.

**Reply by longlong9568** [reply, marked as answer]
I can define it in MainWindow, but I saw in the document that it is possible to define its use in View. What is the reason for this

**Reply by longlong9568** [reply]
What is the reason for this

**Reply by AuroraZiling** [reply]
Declaring this code inside Views won't throw an exception, but also won't have any effect. Because Dialog and Toast components can only be placed inside Hosts, which only exists in SukiWindow and is a property of SukiWindow. Other Views will not have this property in general.

**Reply by longlong9568** [reply]
Got it, then I have another question. I need to first pop up a modal dialog box in the asynchronous thread. Only after the user clicks OK can the thread run the next one. How do I proceed? I just entered MainWindow in the thread DialogManager.CreateDialog()
.WithTitle("Example Dialog")
.WithContent("The content of an example dialog can be seen here.")
.TryShow();  Use this error message
